main_window/rocket_data.py

```diff
@@ -226,16 +229,13 @@ def save(self, csv_path):
 
             data = np.empty((len(keys), len(self.timeset) + 1), dtype=object)
             times = list(self.timeset.keys())
-            times.sort(reverse=False)
             for ix, iy in np.ndindex(data.shape):
                 # Make the first row a list of sensor names. Use the enum's name property
                 if iy == 0:
                     data_name = keys[ix].data_id.name if isinstance(keys[ix].data_id, DataEntryIds) else str(keys[ix].data_id)
                     device = self.device_manager.get_device_type(keys[ix].full_address)
-                    if device:
-                        device_name = device.name
-                    else:
-                        device_name = f"{keys[ix].full_address.connection_name}_{keys[ix].full_address.device_address}"
+                    device_name = device.name if device else \
+                        f"{keys[ix].full_address.connection_name}_{keys[ix].full_address.device_address}"
 
                     data[ix, iy] = data_name + '_' + device_name
                 else:
```

Comment:
ok this isn't strictly part of the changed code, but aren't the `else: if: ` blocks below just `elif:` ?